File: src/designpatterns/builder/BuilderJava/VeganBuilder.java
package designpatterns.builder.BuilderJava;

public class VeganBuilder extends Builder {

    @Override
    public Builder addStarter() {
        meal.setStart(Starter.SALAD);
        return this;
    }

    @Override
    public Builder addMain() {
        meal.setMain(Main.VEGGIE_STIR_FRY);
        return this;
    }

    @Override
    public Builder addDessert() {
        meal.setDessert(Dessert.VEGAN_PUDDING);
        return this;
    }

    @Override
    public Builder addDrink() {
        meal.setDrink(Drink.VEGAN_SHAKE);
        return this;
    }
}

File: src/designpatterns/builder/vegan_builder.cpp
#include "vegan_builder.hpp"
#include "meal.hpp"

Builder* VeganBuilder::addStarter() {
    meal.setStart(SALAD);
    return this;
}

Builder* VeganBuilder::addMain() {
    meal.setMain(VEGGIE_STIR_FRY);
    return this;
}

Builder* VeganBuilder::addDessert() {
    meal.setDessert(VEGAN_PUDDING);
    return this;
}

Builder* VeganBuilder::addDrink() {
    meal.setDrink(VEGAN_SHAKE);
    return this;
}